ceremonies/src/lib.rs

```diff
@@ -383,11 +384,98 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			cid: CommunityIdentifier,
 		) -> DispatchResultWithPostInfo {
-			let sender = ensure_signed(origin)?;
-			let (meetup_index, reward_count) =
-				Self::validate_one_meetup_and_issue_rewards(&sender, &cid)?;
+			let participant = &ensure_signed(origin)?;
+
+			if <encointer_scheduler::Pallet<T>>::current_phase() != CeremonyPhaseType::Registering {
+				return Err(<Error<T>>::WrongPhaseForClaimingRewards.into())
+			}
+
+			let cindex = <encointer_scheduler::Pallet<T>>::current_ceremony_index() - 1; //safe; cindex comes from within
+			let meetup_index = Self::get_meetup_index((cid, cindex), participant)
+				.ok_or(<Error<T>>::ParticipantIsNotRegistered)?;
+
+			if <IssuedRewards<T>>::contains_key((cid, cindex), meetup_index) {
+				return Err(<Error<T>>::RewardsAlreadyIssued.into())
+			}
+			info!(
+				target: LOG,
+				"validating meetup {:?} for cid {:?} triggered by {:?}",
+				meetup_index,
+				&cid,
+				participant
+			);
+
+			//gather all data
+			let meetup_participants = Self::get_meetup_participants((cid, cindex), meetup_index)
+				.ok_or(<Error<T>>::GetMeetupParticipantsError)?;
+			let (participant_votes, participant_attestations) =
+				Self::gather_meetup_validation_data(cid, cindex, meetup_participants.clone());
+
+			// initialize an array of local participant indices that are eligible for the reward
+			// indices will be deleted in the following based on various rules
+			let mut participants_eligible_for_rewards: Vec<usize> =
+				(0..meetup_participants.len()).collect();
+
+			let attestation_threshold_fn = |i: usize| max(i.saturating_sub(1), 1);
+			let participant_judgements = match get_participant_judgements(
+				&participants_eligible_for_rewards,
+				&participant_votes,
+				&participant_attestations,
+				attestation_threshold_fn,
+			) {
+				Ok(participant_judgements) => participant_judgements,
+				// handle errors
+				Err(err) => match err {
+					MeetupValidationError::BallotEmpty => {
+						debug!(
+							target: LOG,
+							"ballot empty for meetup {:?}, cid: {:?}", meetup_index, cid
+						);
+						return Err(<Error<T>>::VotesNotDependable.into())
+					},
+					MeetupValidationError::NoDependableVote => {
+						debug!(
+							target: LOG,
+							"ballot doesn't reach dependable majority for meetup {:?}, cid: {:?}",
+							meetup_index,
+							cid
+						);
+						return Err(<Error<T>>::VotesNotDependable.into())
+					},
+				},
+			};
+			participants_eligible_for_rewards = participant_judgements.legit;
+			// emit events
+			participant_judgements.excluded.iter().for_each(|p| {
+				let participant = meetup_participants[p.index].clone();
+				let event = match p.reason {
+					ExclusionReason::NoVote => {
+						debug!(target: LOG, "participant {:?} received no rewards because they did not vote. meetup_index: {:?}, cid: {:?} cindex: {:?}", participant, meetup_index, cid, cindex);
+						Event::NoRewardBecauseDidNotVote
+					},
```

Comment:
I think we don't need to do this match statement here (see comment below). Then we can also just use one debug statement, with the same text, we print the specific reason anyhow.